_overviews/core/futures.md

```diff
@@ -1121,11 +1122,88 @@ behind this is to prevent propagation of critical and control-flow related
 exceptions normally not handled by the client code and at the same time inform
 the client in which future the computation failed.
 
-Fatal exceptions (as determined by `NonFatal`) are rethrown in the thread executing
+Fatal exceptions (as determined by `NonFatal`) are rethrown from the thread executing
 the failed asynchronous computation. This informs the code managing the executing
 threads of the problem and allows it to fail fast, if necessary. See
 [`NonFatal`](https://www.scala-lang.org/api/current/scala/util/control/NonFatal$.html)
-for a more precise description of the semantics.
+for a more precise description of which exceptions are considered fatal.
+
+`ExecutionContext.global` handles fatal exceptions by printing a stack trace, by default.
+
+A fatal exception means that the `Future` associated with the computation will never complete.
+That is, "fatal" means that the error is not recoverable for the `ExecutionContext`
+and is also not intended to be handled by user code. By contrast, application code may
+attempt recovery from a "failed" `Future`, which has completed but with an exception.
+
+An execution context can be customized with a reporter that handles fatal exceptions.
+See the factory methods [`fromExecutor`](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext$.html#fromExecutor(e:java.util.concurrent.Executor,reporter:Throwable=%3EUnit):scala.concurrent.ExecutionContextExecutor)
+and [`fromExecutorService`](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext$.html#fromExecutorService(e:java.util.concurrent.ExecutorService,reporter:Throwable=%3EUnit):scala.concurrent.ExecutionContextExecutorService).
+
+Since it is necessary to set the [`UncaughtExceptionHandler`](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Thread.UncaughtExceptionHandler.html)
+for executing threads, as a convenience, `fromExecutor` will create a correctly configured context when passed a `null` executor.
+
+The following example demonstrates how to obtain an `ExecutionContext` with custom error handling
+and also shows the result of different exceptions, as described above:
+
+{% tabs exceptions %}
+{% tab 'Scala 2 and 3' for=exceptions %}
```

Comment:
The code is not idiomatic for Scala 3, it would be great to use distinct tabs for Scala 2 and Scala 3 here once we settle on the content.

Reply:
I ran out of steam. Because I removed the non-local example (which didn't make sense), the only difference is the import style, and maybe braces for locally. Thanks for the review, I tried to take all the suggestions, as usual, any remaining errata are my own.

Reply:
Thank you for all the changes so far. I would like to still make more suggestions by directly pushing commits to your branch. I should get some time for that next week.

Reply:
Thanks for making it useful if that's possible.